### PR TITLE
Enhanced CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by MattTheGeek**
More could be changed and improved to the document, but i think this is a good starting point. More rules should be added, and the reasons for those rules and conventions should be made clear. Perhaps the wiki page may be a better place for these changes.

**Change log:**
Rewrote introduction.
Corrected some markup issues.
Corrected typos, and rewrote some rules for better clarity. The rules meaning have not been changed.
Reordered rules into logical groups, and changed rules from a unordered list to level 4 subheaders.
Rewrote and formatted example headers.
Correct some examples, and added additional examples.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -1,141 +1,180 @@
-#Contributing
+# Contributing
 
-Your pull requests are welcome; however, they may not be accepted for various reasons.
+The Notepad++ project welcomes your contributions to the project. However, not all contributions will or can be accepted. Only after careful review and deliberation, will a contribution be accepted.
```
Comment (MattTheGeek) [inline CONTRIBUTING.md]: It just sounds better, doesn't it?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -1,141 +1,180 @@
...
+## Guidelines for Pull Requests
+Although not every contribution can be accepted and merged into the Notepad++ project, Following these guidelines can help get your contribution accepted.
 
-1. Respect Notepad++ coding style.
-2. Make a single change per commit.
-3. Make your modification compact - don't reformat source code in your request. It makes code review more difficult.
-4. PR of reformating (changing of ws/TAB, line endings or coding style) of source code won't be accepted. Use issue trackers for your request instead.
-5. Only bug-fix and feature request will be accepted. For the other things, I can run static code analysis tool myself.
+- Respect the Notepad++ coding style.
+- Make a single change per commit.
```
Comment (MattTheGeek) [inline CONTRIBUTING.md]: This needs to be expanded upon, as it's ambiguous.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -1,141 +1,180 @@
...
+	  default:
+	      // Do something else
+  } // No semi-colon here
+```
+
+### Spacing
 
-* Always leave one space before and after binary and ternary operators
-  Only leave one space after semi-colons in "for" statements.
+#### Always insert one space before and after any binary, logical, or ternary operators.
```
Comment (MattTheGeek) [inline CONTRIBUTING.md]: Added logical operators, as your example has one.

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -1,141 +1,180 @@
...
-      startThermoNuclearWar();
-```
-
-* Prefer enums for integer constants
-
-
+### Naming Conventions
 
-####NAMING CONVENTIONS
+#### Class names should be in PascalCase.
```
Comment (MattTheGeek) [inline CONTRIBUTING.md]: Changed camelCase to PascalCase, as it's what is shown in the examples.

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -144,56 +183,35 @@ GOOD:
...
 ```c
   if (a < 24 && b < 60 && c < 60)
 ```
 
-####COMMENTS
+###Comments
 
-* Use C++ comment line style than c comment style
+#### Use single line comments only.
+Use single line comments only in the project's source code. Use single line comments even if your comment spans for more than one line.
```
Comment (MattTheGeek) [inline CONTRIBUTING.md]: I don't think that C++ needs to be specified, but that's just me.

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -144,56 +183,35 @@ GOOD:
...
-```
+**Bad comment style:**
+```c
   /*
   Please don't piss me off with that
   */
 ```
-
-
-####BEST PRACTICES
```
Comment (MattTheGeek) [inline CONTRIBUTING.md]: Moved "best practices" to the "General" section.

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -1,141 +1,180 @@
...
+#### Avoid the use of magic numbers.
+
+**Bad Example:**
+```c
+  while (lifeTheUniverseAndEverything != 42)
+      lifeTheUniverseAndEverything =  buildMorePowerfulComputerForTheAnswer();
+```
+**Good Example:**
+```c
+  if (foo < iCanPushOnTheBigRedButton)
```
Comment (MattTheGeek) [inline CONTRIBUTING.md]: Changed the good example to follow the project's coding style.

**Comment by donho** [comment]
Let's focus on code enhancement (bug-fix/adding feature) please.

**Comment by ariccio** [comment]
> Let's focus on code enhancement (bug-fix/adding feature) please.

This is indirectly part of code enhancement, as contribution standards/guidelines ease the actual process of contribution.

While I don't think that this pull request is perfect, it seems mostly reasonable...

**Pull request closed without merging**